h2o-algos/src/main/java/hex/anovaglm/AnovaGLMUtils.java

```diff
@@ -0,0 +1,454 @@
+  }
+
+  /***
+   * Simple method to extract GLM Models from GLM ModelBuilders.
+   * @param glmResults: array of GLM ModelBuilders
+   * @return: array of GLMModels
+   */
+  public static GLMModel[] extractGLMModels(GLM[] glmResults) {
+    int numberModel = glmResults.length;
+    GLMModel[] models = new GLMModel[numberModel];
+    for (int index = 0; index < numberModel; index++) {
+      models[index] = glmResults[index].get();
+      Scope.track_generic(models[index]);
+    }
+    return models;
+  }
+  
+  public static void removeFromDKV(Frame[] trainingFrames, int numFrame2Delete) {
+    for (int index=0; index < numFrame2Delete; index++)
+      DKV.remove(trainingFrames[index]._key);
+  }
+
+  /***
+   * This method is used to attach the weight/offset columns if they exist and the response columns, specific 
+   * transformed columns to a training frames.
+   * 
+   * @param predNums: number of all predictor combos
+   * @param allCols: Frame containing all transformed columns
+   * @param transformedColNames: transformed predictor combo arrays containing only predictor combos for a specific
+   *                           training dataset.  Recall that models are built with one predictor combo left out.  This
+   *                           is to generate that training frame with a specific predictor combo left out.
+   * @param parms: AnovaGLMParameters
+   * @return training frame excluding a specific set of predictor combos.
+   */
+  public static Frame buildSpecificFrame(int[] predNums, Frame allCols, String[][] transformedColNames, 
+                                         AnovaGLMParameters parms) {
+    final Frame predVecs = new Frame(Key.make());
+    int numVecs = predNums.length;
+    for (int index = 0; index < numVecs; index++) {
+      int predVecNum = predNums[index];
+      predVecs.add(allCols.subframe(transformedColNames[predVecNum]));
+    }
+    if (parms._weights_column != null)
+      predVecs.add(parms._weights_column, allCols.vec(parms._weights_column));
+    if (parms._offset_column != null)
+      predVecs.add(parms._offset_column, allCols.vec(parms._offset_column));
+    predVecs.add(parms._response_column, allCols.vec(parms._response_column));
+    return predVecs;
+  }
+  
+  public static GLMParameters[] buildGLMParameters(Frame[] trainingFrames, AnovaGLMParameters parms) {
+    int numberOfModels = trainingFrames.length;
+    GLMParameters[] glmParams = new GLMParameters[numberOfModels];
+    List<String> anovaGLMOnlyList = Arrays.asList("save_transformed_framekeys", "type");
+    for (int index = 0; index < numberOfModels; index++) {
+      glmParams[index] = new GLMParameters();
+      Field[] field1 = AnovaGLMParameters.class.getDeclaredFields();
+      setParamField(parms, glmParams[index], false, field1, anovaGLMOnlyList);
+      Field[] field2 = Model.Parameters.class.getDeclaredFields();
+      setParamField(parms, glmParams[index], true, field2, Arrays.asList());
```

Comment:
I would suggest using `Collections.emptyList()` for an empty list creation

Reply:
@koniecsveta :  I was not trying to create an empty list here, rather, I am trying to get the fields in AnovaGLMParameters and Model.Parameters.  Next, I will copy these parameter values into the GLMParameters.  However, I did move field1 and field 2 outside the for-loop since these two values do not change.

Reply:
I meant replacing `Arrays.asList()` by `Collections.emptyList()`

Reply:
Yes, done!  I was being super thick here.